pkg/apis/kubeflow.org/v1/mpi_defaults.go

```diff
@@ -32,9 +32,10 @@ func SetDefaults_MPIJob(mpiJob *MPIJob) {
 		mpiJob.Spec.RunPolicy.CleanPodPolicy = &none
 	}
 
-	// set default to Launcher
-	setDefaultRestartPolicy(mpiJob.Spec.MPIReplicaSpecs[MPIJobReplicaTypeLauncher], MPIJobDefaultRestartPolicy)
-
-	// set default to Worker
-	setDefaultRestartPolicy(mpiJob.Spec.MPIReplicaSpecs[MPIJobReplicaTypeWorker], MPIJobDefaultRestartPolicy)
+	for _, spec := range mpiJob.Spec.MPIReplicaSpecs {
+		// Set default replicas to 1.
+		setDefaultReplicas(spec, 1)
```

Comment:
```suggestion
		mpiJob.Spec.MPIReplicaSpecs[MPIJobReplicaTypeLauncher].Replicas = pointer.Int32(1)
```

We should enforce 1 to `replicas` for the Launcher since we don't support launching multiple Lunacher.

Reply:
Also, I don't think we need defaulting 1 to replicas for the Worker since we support workers elastically.
So, it is enough to set 0 to replicate in the Worker iff the `replicas` isn't set.

Reply:
> We should enforce 1 to `replicas` for the Launcher since we don't support launching multiple Lunacher.

Users may feel confused because in the CRD, it is a customizable value. Setting the default value to 1 when replicas is nil may be better (as done in mpi-operator v2), WDYT?

> it is enough to set 0 to replicate in the Worker iff the replicas isn't set.

agree

Reply:
> Users may feel confused because in the CRD, it is a customizable value. Setting the default value to 1 when replicas is nil may be better (as done in mpi-operator v2), WDYT?

Yes, I meant that. sgtm.